src/object_store/src/object/s3.rs

```diff
@@ -723,6 +722,7 @@ impl S3ObjectStore {
                 bucket,
                 configured_rules,
             );
+            true
```

Comment:
As indicated by the log line in L721, reaching here doesn't mean the bucket retention is enabled. Instead, it only means that the bucket has already configured the lifecycle rule to expire incomplete multipart upload and this should not prevent the cluster from starting up.

After skimming through the sdk codes, I think we should check `rule.expiration().is_some()` in L710.

Reply:
Indeed.